sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala

```diff
@@ -241,8 +241,14 @@ private[hive] object HiveShim {
       Decimal(hdoi.getPrimitiveJavaObject(data).bigDecimalValue())
     }
   }
+
+  implicit def prepareWritable(shimW: ShimWritable): Writable = {
+    shimW.writable
+  }
 }
 
+case class ShimWritable(writable: Writable)
```

Comment:
Why do we need `ShimWriteable`?  Couldn't the `prepareWriteable` function just take a `Writeable`?  Furthermore, does it break Hive 12 if we just always fix avro writeable?  It would be good to minimize the size of the shim when possible.